packages/components/src/components/post-tooltip/post-tooltip.tsx

```diff
@@ -43,6 +43,19 @@ export class PostTooltip {
     );
   }
 
+  private extractText(node: Node): string {
+    if (node.nodeType === Node.TEXT_NODE) {
+      return node.textContent?.trim() ?? '';
+    }
+    if (node.nodeType === Node.ELEMENT_NODE) {
+      const element = node as Element;
+      return Array.from(element.childNodes)
+        .map(child => this.extractText(child))
+        .join(' ');
+    }
+    return '';
```

Comment:
Imo a switch statement would fit your needs 100% here and also increase readability and performence.
https://developer.mozilla.org/de/docs/Web/JavaScript/Reference/Statements/switch

But also, where is this method used, I couldn't find any execution of it?